tests/Query/Filter/PriorityField.test.ts

```diff
@@ -0,0 +1,81 @@
+    filter: string,
+    priority: Priority,
+    expected: boolean,
+) {
+    const builder = new TaskBuilder();
+    const filterOrError = new PriorityField().createFilterOrErrorMessage(
+        filter,
+    );
+    testTaskFilter(filterOrError, builder.priority(priority).build(), expected);
+}
+
+describe('priority is', () => {
+    it('priority is high', () => {
+        const filter = 'priority is high';
+        testTaskFilterForTaskWithPriority(filter, Priority.High, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.Medium, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, false);
+    });
+
+    it('priority is medium', () => {
+        const filter = 'priority is medium';
+        testTaskFilterForTaskWithPriority(filter, Priority.High, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Medium, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, false);
+    });
+
+    it('priority is none', () => {
+        const filter = 'priority is none';
+        testTaskFilterForTaskWithPriority(filter, Priority.High, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Medium, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, false);
+    });
+
+    it('priority is low', () => {
+        const filter = 'priority is low';
+        testTaskFilterForTaskWithPriority(filter, Priority.High, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Medium, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, true);
+    });
+});
+
+describe('priority above', () => {
+    it('priority above none', () => {
+        const filter = 'priority above none';
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Medium, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.High, true);
+    });
+});
+
+describe('priority below', () => {
+    it('priority below none', () => {
+        const filter = 'priority below none';
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, false);
+    });
+});
+
+describe('priority error cases', () => {
+    it('priority is something', () => {
+        const field = new PriorityField();
+        const filter = field.createFilterOrErrorMessage(
+            'priority is no-such-priority',
+        );
+        expect(filter.filter).toBeUndefined();
+        expect(filter.error).toBe(
+            'do not understand query filter (priority date)',
```

Comment:
This text is wrong - the ' date' must have been copied in from another filter. Will fix as a separate commit.